temporal-sdk/src/main/java/io/temporal/workflow/DynamicWorkflow.java

```diff
@@ -0,0 +1,45 @@
+/*
+ *  Copyright (C) 2020 Temporal Technologies, Inc. All Rights Reserved.
+ *
+ *  Copyright 2012-2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Modifications copyright (C) 2017 Uber Technologies, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not
+ *  use this file except in compliance with the License. A copy of the License is
+ *  located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed on
+ *  an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+package io.temporal.workflow;
+
+import io.temporal.common.converter.EncodedValues;
+
+/**
+ * Use DynamicWorkflow to implement any number of workflow types dynamically. When a workflow
+ * implementation type that extends DynamicWorkflow is registered, it is used to implement any
+ * workflow type that is not implicitly registered with the {@link io.temporal.worker.Worker}. Only
+ * one type that implements DynamicWorkflow per worker is allowed.
+ *
+ * <p>The main use case for DynamicWorkflow is an implementation of custom Domain Specific Languages
+ * (DSLs). A single implementation can implement a workflow type which definition is dynamically
+ * loaded from some external source.
+ *
```

Comment:
Presumably it's necessary for the dynamic workflow to always be deterministic in the same way that normal workflows are, right? IE: Deterministic w.r.t. `EncodedValues`. Might be worth making reference to that here just so users don't think magically that requirement has gone away.